src/realm/sync/noinst/client_history_impl.cpp

```diff
@@ -947,6 +948,19 @@ void ClientHistory::update_sync_progress(const SyncProgress& progress, Downloada
     root.set(s_progress_uploaded_bytes_iip,
              RefOrTagged::make_tagged(uploaded_bytes)); // Throws
 
+    if (previous_upload_client_version < progress.upload.client_version) {
```

Comment:
does this assume that if we make any upload progress that we'll have fully uploaded all changes and we know for sure we aren't going to get another client reset from any recovered changesets? maybe now that we have compensating writes that doesn't really matter as much?

Reply:
Uploading changesets should *either* result in the server acknowledging the upload or sending a client reset and not both, so once our UPLOAD is acked the window for getting a client reset due to those changesets being invalid has ended.

I think this check is probably wrong and it needs to actually be checking if we've reached the client version at the time of the client reset (or at time of opening).

Reply:
It took a while to figure out how to test it but this is indeed incorrect; it marks the client reset as complete as soon as *any* changesets are acked rather than when all of the recovered changesets are.